packages/quantic/force-app/main/default/lwc/quanticCaseClassification/e2e/quanticCaseClassification.e2e.ts

```diff
@@ -0,0 +1,269 @@
+          caseClassification.waitForTicketFieldUpdateCollectAnalytics({
+            fieldName: coveoDefaultField,
+          });
+
+        await caseClassification.clickShowSelectInputButton();
+        await caseClassification.clickallOptionsSelectInput();
+        await caseClassification.clickSelectInputOptionAtIndex(3);
+        await fieldUpdateCollectEventPromise;
+      });
+    });
+  });
+
+  test.describe('when the property fetchClassificationOnChange is set to true', () => {
+    test.use({
+      options: {fetchClassificationOnChange: true},
+    });
+
+    test('should trigger a new classification request when a case classification suggestion is clicked', async ({
+      caseClassification,
+      caseAssist,
+    }) => {
+      const caseClassificationsResponsePromise =
+        caseAssist.waitForCaseClassificationsResponse();
+
+      await caseClassification.clickCaseClassificationSuggestionAtIndex(1);
+
+      await caseClassificationsResponsePromise;
+    });
+
+    test('should trigger a new classification request when a case classification option is clicked', async ({
+      caseClassification,
+      caseAssist,
+    }) => {
+      const caseClassificationsResponsePromise =
+        caseAssist.waitForCaseClassificationsResponse();
+
+      await caseClassification.clickShowSelectInputButton();
+      await caseClassification.clickallOptionsSelectInput();
+      await caseClassification.clickSelectInputOptionAtIndex(3);
+
+      await caseClassificationsResponsePromise;
+    });
+  });
+
+  test.describe('when the property fetchDocumentSuggestionOnChange is set to true', () => {
+    test.use({
+      options: {fetchDocumentSuggestionOnChange: true},
+    });
+
+    test('should trigger a new document suggestions request when a case classification suggestion is clicked', async ({
+      caseClassification,
+      caseAssist,
+    }) => {
+      const documentSuggestionResponsePromise =
+        caseAssist.waitForDocumentSuggestionResponse();
+
+      await caseClassification.clickCaseClassificationSuggestionAtIndex(1);
+
+      await documentSuggestionResponsePromise;
+    });
+
+    test('should trigger a new document suggestions request when a case classification option is clicked', async ({
+      caseClassification,
+      caseAssist,
+    }) => {
+      const documentSuggestionResponsePromise =
+        caseAssist.waitForDocumentSuggestionResponse();
+
+      await caseClassification.clickShowSelectInputButton();
+      await caseClassification.clickallOptionsSelectInput();
+      await caseClassification.clickSelectInputOptionAtIndex(3);
+
+      await documentSuggestionResponsePromise;
+    });
+  });
+
+  test.describe('the behaviour of the case classifications', () => {
+    test.describe('when selecting a suggestion and then receiving new suggestions that include the selected suggestion', () => {
+      test('should keep the suggestion by the user selected and display it as an inline option', async ({
```

Comment:
This is pretty close to something that should be better tested as a unit test no?

Reply:
answered here: https://github.com/coveo/ui-kit/pull/4967#discussion_r1973683527

Reply:
Answered in the other comment, I'll resolve this one.